src/clusterfuzz/_internal/system/new_process.py

```diff
@@ -414,36 +421,37 @@ class UnicodeProcessRunner(UnicodeProcessRunnerMixin, ProcessRunner):
   """ProcessRunner which always returns unicode output."""
 
 
-class UnshareProcessRunnerMixin(object):
-  """ProcessRunner mixin which unshares."""
+class ModifierProcessRunnerMixin(object):
+  """ProcessRunner mixin with modifiers."""
+
+  def tool_prefix(self, tool):
+    """Prefix the command with a tool and its args"""
+    if not environment.get_value(f'USE_{tool.upper()}'):
+      return []
+
+    tool_path = environment.get_default_tool_path(tool)
```

Comment:
We'll also need to add a mechanism to download the binary if it is not installed in our image. e.g. for our "extra_sanitizers" we can hardcode it to a GCS bucket path to be downloaded as needed.